src/main/java/frc/robot/subsystems/swerve/SwerveSubsystem.java

```diff
@@ -81,6 +81,21 @@ public SwerveSubsystem() {
      * @param angularPower The angular (rotational) power [-1.0, 1.0].
      */
     public void setSwerveDrivePowers(double xPower, double yPower, double angularPower) {
+        // If drivers are sending no input, stop all modules but hold their current angle.
+        if (xPower == 0.0 && yPower == 0.0 && angularPower == 0.0) {
+            SwerveModuleState tlState = topLeftModule.getState();
+            SwerveModuleState trState = topRightModule.getState();
+            SwerveModuleState blState = bottomLeftModule.getState();
+            SwerveModuleState brState = bottomRightModule.getState();
+
+            this.states[0] = new SwerveModuleState(0.0, tlState.angle);
```

Comment:
I also just realized that we can use the previous setpoint's angle (`states[0].angle`) instead of the queried angle to prevent one call to the encoder. Is this better?